be/src/runtime/statistic_result_writer.cpp

```diff
@@ -79,7 +79,6 @@ Status StatisticResultWriter::append_chunk(vectorized::Chunk* chunk) {
     }
 
     LOG(WARNING) << "Append statistic result to sink failed.";
-    delete result;
```

Comment:
Is previous code exception-safe? Does all code path release the `result` ?

Reply:
> Is previous code exception-safe? Does all code path release the `result` ?

In previous logic, `result` will be released if it's added into `_sinker` successfully, otherwise, it should be deleted here. I think using raw pointer is not convenient, so I change it to unique_ptr.